src/EscapeCodeGenerator.hpp

```diff
@@ -0,0 +1,47 @@
+#pragma once
+
+#include "Defs.hpp"
+#include <string>
+
+struct EscapeCodeGenerator {
+
+    using EscapeCode = std::string;
+
+    virtual EscapeCode cursorForward(int) { return EscapeCode{}; }
+    virtual EscapeCode cursorBackward(int) { return EscapeCode{}; }
+    virtual EscapeCode clearTerminalLine() { return EscapeCode{}; }
+    virtual EscapeCode setColor(Color) { return EscapeCode{}; }
+
```

Comment:
yes, I understand, but in the future someone will use it the other way around and we might have been screwed up by that time :P